core/src/test/java/com/linecorp/armeria/server/conditional/ConditionalRequestUtilTest.java

```diff
@@ -0,0 +1,145 @@
+import static com.linecorp.armeria.server.conditional.ConditionalRequestUtil.strongComparison;
+import static com.linecorp.armeria.server.conditional.ConditionalRequestUtil.weakComparison;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Instant;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.RequestHeaders;
+
+class ConditionalRequestUtilTest {
+    private static final Long LONG_TIME_AGO = Instant.parse("2007-12-03T10:15:30.00Z").toEpochMilli();
+    private static final Long NOT_SO_LONG_TIME_AGO = Instant.parse("2008-12-03T10:15:30.00Z").toEpochMilli();
+
+    /*
+     * Comparison table:
+     * ETag 1       ETag 2      Strong Comparison   Weak Comparison
+     * W/"1"        W/"1"       no match            match
+     * W/"1"        W/"2"       no match            no match
+     * W/"1"        "1"         no match            match
+     * "1"          "1"         match               match
+     */
+    @Test
+    void testWeakComparison() {
+        assertThat(weakComparison(ImmutableList.of(new ETag("1", true)), new ETag("1", true)))
+                .isTrue();
+        assertThat(weakComparison(ImmutableList.of(new ETag("1", true)), new ETag("2", true)))
+                .isFalse();
+        assertThat(weakComparison(ImmutableList.of(new ETag("1", false)), new ETag("1", true)))
+                .isTrue();
+        assertThat(weakComparison(ImmutableList.of(new ETag("1", false)), new ETag("1", false)))
+                .isTrue();
+    }
+
+    @Test
+    void testStrongComparison() {
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", true)), new ETag("1", true)))
+                .isFalse();
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", true)), new ETag("2", true)))
+                .isFalse();
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", false)), new ETag("1", true)))
+                .isFalse();
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", false)), new ETag("1", false)))
+                .isTrue();
+    }
+
+    @Test
+    void testIfMatch() {
+        assertThat(ConditionalRequestUtil.ifMatch("W/\"1\"", new ETag("1", true)))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
+        assertThat(ConditionalRequestUtil.ifMatch("*", new ETag("1", true)))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
+        assertThat(ConditionalRequestUtil.ifMatch("*", null))
+                .isSameAs(ETagResponse.SKIP_METHOD_PRECONDITION_FAILED);
+        assertThat(ConditionalRequestUtil.ifMatch("\"1\"", new ETag("1", true)))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
+        assertThat(ConditionalRequestUtil.ifMatch("\"1\"", new ETag("1", false)))
+                .isSameAs(ETagResponse.SKIP_METHOD_PRECONDITION_FAILED);
```

Comment:
Strong matching Strong - should run

```suggestion
                .isSameAs(ETagResponse.PERFORM_METHOD);
```